rust/src/frames.rs

```diff
@@ -100,13 +99,6 @@ impl Frame {
         };
     }
 
-    #[allow(clippy::not_unsafe_ptr_arg_deref)]
-    pub fn set_tx(&self, flow: *const Flow, tx_id: u64) {
-        unsafe {
-            AppLayerFrameSetTxIdById(flow, self.direction(), self.id, tx_id);
-        };
-    }
-
```

Comment:
Shouldn't we leave this, as we're planning to expand the usage and coverage of frames for other protocols?

Reply:
There are argument both ways.

Pro-remove: Unused code is untested and is or will get broken soon

Reply:
Ok, the only thing I can think of for keeping it would be that if we don't have that, it is less likely that people adding frames will think of using something like that?
Would using `#[allow(dead_code)]` be acceptable? I'm guessing not, right.

Reply:
Making these `public` makes them not dead_code, even if unused.
We should make most `pub(crate)` or even `pub(self)`